### PR TITLE
Surreals 4

### PR DESCRIPTION
**Pull request by sctfn**
I'm afraid that I got a bit lost in the weeds with induction here. Here's some new material, though. I also added some new $v declarations - a wff al and setvars xO yO xL yL xR and yR

### CODE REVIEW

**Comment by icecream17** [comment]
(The first 24 commits are duplicates of #4146 due to `git` shenanigans, the first new commit is [(prevent github from converting the link because this version has the nice "prev" and "next" buttons) `9a32f7d`](https://github.com/metamath/set.mm/pull/4155/commits/9a32f7d8ec36c4cbd17485fbd2eb231643d0c0f1))

At commit `076621a` github does not have a next button, so [link `bdda652`](https://github.com/metamath/set.mm/pull/4155/commits/bdda652693dbda9b4f55d0957ba93e5341b51b3e)

File: set.mm [hunk 1]
```diff
@@ -28607,6 +28610,7 @@ generally appear in a single form (either definitional, but more often
       ( cv wcel wi ex a2d ralimdv2 ) ABCDEEADGEHZBCAMBCIFJKL $.
   $}
 
+
```
Comment (icecream17) [inline set.mm:28613]: extra line

Now that I think about it, ~ conventions doesn't say anything about whitespace, the only thing is [`CONTRIBUTING.md`: following metamath.exe rewrap](https://github.com/metamath/set.mm/blob/develop/CONTRIBUTING.md#formatting-recommendation-prior-to-submitting-a-pull-request), which doesn't detect this. The usual spacing seems to be one line between items like theorems or blocks, and two lines at the end of each section, but there are plenty of counterexamples...

File: set.mm [hunk 2]
```diff
@@ -522751,6 +522816,135 @@ Real and complex numbers (cont.)
...
+      ( cv c2nd cfv wsbc c1st nfsbc1v nfcv nfsbcw nfv sbcopeq1a ralxpf ) ADBGZH
+      IZJZCRKIZJABCDEFTCUALTDCUADUAMADSLNABOACDRPQ $.
+  $}
+
+  $( Ordered triple membership in a triple cross product.  (Contributed by
+     Scott Fenton, 21-Aug-2024.) $)
+  otelxp $p |- ( <. <. A , B >. , C >. e. ( ( D X. E ) X. F ) <->
+  ( A e. D /\ B e. E /\ C e. F ) ) $=
+    ( cop cxp wcel wa w3a opelxp anbi1i df-3an 3bitr4i ) ABGZDEHZIZCFIZJADIZBEI
+    ZJZSJPCGQFHITUASKRUBSABDELMPCQFLTUASNO $.
```
Comment (icecream17) [inline set.mm:522836]: [df-ot](https://us.metamath.org/mpeuni/df-ot.html) exists, so these theorems could conflict in the future (especially ot1std and ot2ndd), although df-ot isn't used very much.

Running `show usage df-ot /rec` I get `Statement "df-ot" directly or indirectly affects the proofs of 774 statements`, about 600 or so coming from theorems that use [ot1stg](https://us.metamath.org/mpeuni/ot1stg.html)

Perhaps the name could be "o3" or "ot2"? Not sure, perhaps others have some ideas.

File: set.mm [hunk 3]
```diff
@@ -31526,6 +31530,26 @@ generally appear in a single form (either definitional, but more often
...
+  ${
+    ralimda.1 $e |- F/ x ph $.
+    ralimda.2 $e |- ( ( ph /\ x e. A ) -> ( ps -> ch ) ) $.
+    $( Deduction quantifying both antecedent and consequent.  (Contributed by
+       Glauco Siliprandi, 23-Oct-2021.) $)
+    ralimda $p |- ( ph -> ( A. x e. A ps -> A. x e. A ch ) ) $=
+      ( wral wa nfra1 nfan cv wcel id adantlr rspa adantll sylc ralrimia ex ) A
+      BDEHZCDEHAUAIZCDEAUADFBDEJKUBDLEMZIAUCIZBCAUCUDUAUDNOUAUCBABDEPQGRST $.
+  $}
+
```
Comment (icecream17) [inline set.mm:31552]: move empty line

```suggestion
  ${
    ralrimia.1 $e |- F/ x ph $.
    ralrimia.2 $e |- ( ( ph /\ x e. A ) -> ps ) $.
    $( Inference from Theorem 19.21 of [Margaris] p. 90 (restricted quantifier
       version).  (Contributed by Glauco Siliprandi, 23-Oct-2021.) $)
    ralrimia $p |- ( ph -> A. x e. A ps ) $=
      ( cv wcel ex ralrimi ) ABCDEACGDHBFIJ $.
  $}

  ${
    ralimda.1 $e |- F/ x ph $.
    ralimda.2 $e |- ( ( ph /\ x e. A ) -> ( ps -> ch ) ) $.
    $( Deduction quantifying both antecedent and consequent.  (Contributed by
       Glauco Siliprandi, 23-Oct-2021.) $)
    ralimda $p |- ( ph -> ( A. x e. A ps -> A. x e. A ch ) ) $=
      ( wral wa nfra1 nfan cv wcel id adantlr rspa adantll sylc ralrimia ex ) A
      BDEHZCDEHAUAIZCDEAUADFBDEJKUBDLEMZIAUCIZBCAUCUDUAUDNOUAUCBABDEPQGRST $.
  $}


```

**Review by icecream17 — APPROVED** [review]
nice shortenings! Personally I think `walpha` should indeed be in main since I once needed to declare more variables too: https://github.com/icecream17/Stuff/blob/main/math/~w2.mm

And the order of variables matters in compressed proofs, so it's best to move it early

File: set.mm (hunk 2 above)
Comment (sctfn) [reply set.mm]: Done - I decided on ot2* for these theorems

File: set.mm (hunk 1 above)
Comment (sctfn) [reply set.mm:28613]: Fair enough. I'll touch this up for now and leave the conversation for another day...

File: set.mm (hunk 3 above)
Comment (sctfn) [reply set.mm]: Fixed

File: set.mm (hunk 1 above)
Comment (benjub) [reply set.mm:28613]: At some point, we agreed on the rule "double blank lines if and only if they separate sections". This rule is globally followed in set.mm and iset.mm.  But this is not done by the automatic rewrapping (wouldn't be too hard to do, probably), so we do not enforce it.

File: set.mm [hunk 4]
```diff
@@ -424,6 +425,8 @@ useful for discussing objects (beyond simply true or false).  In particular,
   wla $f wff la $.
   $( Let variable ` ka ` be a wff. $)
   wka $f wff ka $.
+  $( Let variable ` al ` be a wff. $)
+  walpha $f wff al $.
```
Comment (benjub) [inline set.mm:429]: I remember at some point I proposed to move all such declarations at the beginning of the file, like here, but I was warned about a performance cost for the provers (albeit very small). It's not needed in the main section. Maybe it should be declared at the beginning of your mathbox, like `bnjwphm` and following, and `wjph` and following, for instance.

File: mmset.raw.html [hunk 5]
```diff
@@ -5414,6 +5423,11 @@
 [QA251.5 .G642 1999].
 </LI>
 <LI>
+<A NAME="Goshnor"></A> [Goshnor] Goshnor, Harry, <I>An Introduction to
+the Theory of Surreal Numbers,</I> London Mathematical Societs (1986)
```
Comment (benjub) [inline mmset.raw.html:5427]:
```suggestion
the Theory of Surreal Numbers,</I> London Mathematical Society (1986)
```

Comment (sctfn) [reply mmset.raw.html]: Done

File: set.mm (hunk 4 above)
Comment (sctfn) [reply set.mm]: Done

File: set.mm [hunk 6]
```diff
@@ -31551,6 +31552,25 @@ generally appear in a single form (either definitional, but more often
...
   $}
 
+  ${
+    ralrimia.1 $e |- F/ x ph $.
+    ralrimia.2 $e |- ( ( ph /\ x e. A ) -> ps ) $.
+    $( Inference from Theorem 19.21 of [Margaris] p. 90 (restricted quantifier
+       version).  (Contributed by Glauco Siliprandi, 23-Oct-2021.) $)
+    ralrimia $p |- ( ph -> A. x e. A ps ) $=
+      ( cv wcel ex ralrimi ) ABCDEACGDHBFIJ $.
+  $}
```
Comment (tirix) [inline set.mm:31562]: As a side note, I imagine many theorems can be shortened using `ralrimia`.

File: set.mm [hunk 7]
```diff
@@ -522553,6 +522614,12 @@ Real and complex numbers (cont.)
 #*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#
 $)
 
+  $( Set up a new wff var $)
+  $v al $.  $( Greek alpha $)
+
+  $( Let variable ` al ` be a wff. $)
+  walpha $f wff al $.
+
```
Comment (tirix) [inline set.mm:522622]: I wonder in which order Norm chose to introduce wff variable names. It seems to somehow start from the end of the greek alphabet and go backwards, with some jumps. I'm not sure what would be the next letter to choose.

File: set.mm [hunk 8]
```diff
@@ -522969,6 +523036,135 @@ Real and complex numbers (cont.)
...
+      ( cv c2nd cfv wsbc c1st nfsbc1v nfcv nfsbcw nfv sbcopeq1a ralxpf ) ADBGZH
+      IZJZCRKIZJABCDEFTCUALTDCUADUAMADSLNABOACDRPQ $.
+  $}
+
+  $( Ordered triple membership in a triple cross product.  (Contributed by
+     Scott Fenton, 21-Aug-2024.) $)
+  ot2elxp $p |- ( <. <. A , B >. , C >. e. ( ( D X. E ) X. F ) <->
+  ( A e. D /\ B e. E /\ C e. F ) ) $=
+    ( cop cxp wcel wa w3a opelxp anbi1i df-3an 3bitr4i ) ABGZDEHZIZCFIZJADIZBEI
+    ZJZSJPCGQFHITUASKRUBSABDELMPCQFLTUASNO $.
```
Comment (tirix) [inline set.mm:523056]: Why not use the `<. A , B , C >.` notation for ordered triples? ([df-ot](https://us.metamath.org/mpeuni/df-ot.html))
Granted, it's equivalent.

**Review by tirix — APPROVED** [review]
It's really nice to have all the additional references!

**Comment by benjub** [comment]
Apparently GitHub does not permit to remove oneself as a reviewer, but: I'm "morally" removing myself, since my comments have been addressed and I cannot review rigorously enough to approve.

File: set.mm (hunk 8 above)
Comment (sctfn) [reply set.mm:523056]: Mostly because I forgot about ordered triples. I'm thinking of revisiting these with OTs when I get around to it.

**Pull request merged**